packages/playback-core/src/index.ts

```diff
@@ -323,6 +358,64 @@ export const loadMedia = (
         })
       );
     });
+    hls.on(Hls.Events.NON_NATIVE_TEXT_TRACKS_FOUND, (_type, { tracks }) => {
+      tracks.forEach((trackObj) => {
+        const baseTrackObj = trackObj.subtitleTrack ?? trackObj.closedCaptions;
+        const idx = hls.subtitleTracks.findIndex(({ lang, name, type }) => {
+          return lang == baseTrackObj?.lang && name === trackObj.label && type.toLowerCase() === trackObj.kind;
+        });
+        createTextTrack(
+          mediaEl,
+          trackObj.kind as TextTrackKind,
+          trackObj.label,
+          baseTrackObj?.lang,
+          `${trackObj.kind}${idx}`
+        );
+      });
+    });
+    hls.on(Hls.Events.CUES_PARSED, (_type, { track, type, cues }) => {
+      const textTrack: TextTrack = Array.prototype.find.call(mediaEl.textTracks, (textTrack) => {
+        return textTrack.kind === type && textTrack.id === track;
+      });
+      if (!textTrack) return;
+      const disabled = textTrack.mode === 'disabled';
+      if (disabled) {
+        textTrack.mode = 'hidden';
+      }
+      cues.forEach((cue: VTTCue) => {
+        if (textTrack.cues?.getCueById(cue.id)) return;
```

Comment:
Do we need to see if the cue exists already? Any issues with unconditionally adding the cue into the track? It would allow us not to fiddle with the track's mode.